neard/src/cli.rs

```diff
@@ -455,47 +460,90 @@ impl RunCmd {
+
+            let sys = actix::System::new();
+            let tx_crash = tx.clone();
+            let rx_crash = tx.subscribe();
+            let s = store.clone();
+            let (tx_sig, mut rx_sig) = oneshot::channel::<&str>();
+            let config = near_config.clone();
+
+            let _ = sys.block_on(async move {
+                // Initialize the subscriber that takes care of both logging and tracing.
+                #[cfg(not(feature = "watch_config"))]
+                let _subscriber_guard = default_subscriber_with_opentelemetry(
+                    make_env_filter(_verbose_target).unwrap(),
+                    o11y_opts,
+                    config.client_config.chain_id.clone(),
+                    config.network_config.node_key.public_key().clone(),
+                    config
+                        .network_config
+                        .validator
+                        .as_ref()
+                        .map(|validator| validator.account_id()),
+                )
+                .await
+                .global();
+
+                let nearcore::NearNode { rpc_servers, .. } =
+                    nearcore::start_with_config_and_synchronization(
+                        home_dir,
+                        config,
+                        Some(tx_crash),
+                        Some(s),
+                    )
                     .expect("start_with_config");
 
-            let sig = wait_for_interrupt_signal(home_dir, rx).await;
-            warn!(target: "neard", "{}, stopping... this may take a few minutes.", sig);
-            futures::future::join_all(rpc_servers.iter().map(|(name, server)| async move {
-                server.stop(true).await;
-                debug!(target: "neard", "{} server stopped", name);
-            }))
-            .await;
-            actix::System::current().stop();
-
-            // Disable the subscriber to properly shutdown the tracer.
-            near_o11y::reload(Some("error"), None, Some(OpenTelemetryLevel::OFF)).unwrap();
-        });
-        sys.run().unwrap();
+                let sig = wait_for_interrupt_signal(home_dir, rx_crash).await;
+                futures::future::join_all(rpc_servers.iter().map(|(name, server)| async move {
+                    server.stop(true).await;
+                    debug!(target: "neard", "{} server stopped", name);
+                }))
+                .await;
+                actix::System::current().stop();
+                // Disable the subscriber to properly shutdown the tracer.
+                #[cfg(not(feature = "watch_config"))]
+                near_o11y::reload(Some("error"), None, Some(OpenTelemetryLevel::OFF)).unwrap();
+                tx_sig.send(sig)
+            });
+            sys.run().unwrap();
+            match rx_sig.try_recv() {
+                Ok(sig) => {
+                    if sig == "reload signer" {
+                        info!(target: "neard", "{}, restarting... this may take a few minutes.", sig);
```

Comment:
What can take minutes in the `stopping...` case is shutting down RocksDB, which doesn't happen in the `restarting...` case.